### PR TITLE
[user] blocked users can't delete their account

### LINKED ISSUE #13235: Don't allow blocked users to delete their account
We have the right to keep data of users who have violated our terms (abuse). So we should disable the possibility to delete the user account for blocked users.

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
We want to keep the data of blocked (abusing) users to protect us from them in the future.

## Related Issue(s)

Fixes #13235

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Comment by roboquat** [comment]
@svenefftinge: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -598,7 +598,9 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     public async deleteAccount(ctx: TraceContext): Promise<void> {
         const user = this.checkAndBlockUser("deleteAccount");
         await this.guardAccess({ kind: "user", subject: user! }, "delete");
-
+        if (user.blocked) {
+            throw new ResponseError(ErrorCodes.PERMISSION_DENIED, `You are blocked and can't delete your account.`);
```
Comment (AlexTugarev) [inline components/server/src/workspace/gitpod-server-impl.ts:602]: @svenefftinge, that was already done in https://github.com/gitpod-io/gitpod/pull/9372.
the `this.checkAndBlockUser("deleteAccount")` contains the errors to be thrown if `user.blocked === true`

Comment (svenefftinge) [reply components/server/src/workspace/gitpod-server-impl.ts:602]: Awesome! 🎉

**Pull request closed without merging**